extensions/blocks/google-calendar/edit.js

```diff
@@ -0,0 +1,193 @@
+	};
+
+	setAttributes = event => {
+		if ( event ) {
+			event.preventDefault();
+		}
+		const { editedEmbed } = this.state;
+		const embedString = editedEmbed.trim();
+		let attributes;
+
+		if ( IFRAME_REGEX.test( embedString ) ) {
+			attributes = extractAttributesFromIframe( embedString );
+		} else {
+			attributes = { url: embedString };
+		}
+
+		if ( ! URL_REGEX.test( attributes.url ) ) {
+			this.setErrorNotice();
+			return;
+		}
+
+		this.props.setAttributes( attributes );
+		this.setState( { editingUrl: false, notice: null } );
+	};
+
+	setErrorNotice = () => {
+		this.setState( {
+			notice: __(
+				"Your calendar couldn't be embedded. Please double check your URL or code.",
+				'jetpack'
+			),
+		} );
+	};
+
+	/**
+	 * Render a preview of the Google Calendar embed.
+	 *
+	 * @returns {object} The UI displayed when user edits this block.
+	 */
+	render() {
+		const { attributes, className } = this.props;
+		const { url } = attributes;
+		const { editedEmbed, interactive, editingUrl } = this.state;
+
+		const height = isMobile() ? '300' : '500';
+
+		const html = `<iframe src="${ url }" style="border:0" scrolling="no" frameborder="0" width="100%" height=${ height }></iframe>`;
+
+		const permissionsLink = (
+			<ExternalLink href="https://en.support.wordpress.com/google-calendar/">
+				{ __( 'Enable Permissions for the calender you want to share', 'jetpack' ) }
+			</ExternalLink>
+		);
+
+		const controls = (
+			<>
+				<InspectorControls>
+					<PanelBody title={ __( 'Calendar Settings', 'jetpack' ) } initialOpen={ false }>
+						<form onSubmit={ this.setAttributes } className={ `${ className }-embed-form-sidebar` }>
+							<textarea
+								type="text"
+								value={ editedEmbed }
+								className="components-placeholder__input"
+								aria-label={ __( 'Google Calendar URL or iframe', 'jetpack' ) }
+								placeholder={ __( 'Enter URL or iframe to embed here…', 'jetpack' ) }
+								onChange={ event => this.setState( { editedEmbed: event.target.value } ) }
+							/>
+							<Button isSecondary isLarge type="submit">
+								{ _x( 'Embed', 'button label', 'jetpack' ) }
+							</Button>
+						</form>
+					</PanelBody>
+				</InspectorControls>
+			</>
+		);
+
+		if ( editingUrl || ! url ) {
+			return (
+				<div className={ className }>
+					{ controls }
+					<Placeholder
+						className="wp-block-jetpack-google-calendar"
+						label={ __( 'Google Calendar', 'jetpack' ) }
+						icon={ <BlockIcon icon={ icon } /> }
+						notices={
+							this.state.notice && (
+								<Notice status="error" isDismissible={ false }>
+									{ this.state.notice }
+								</Notice>
+							)
+						}
+					>
+						<p>
+							{ __( 'Step 1)', 'jetpack' ) } { permissionsLink }
+						</p>
+						<p>
+							{ __(
+								'Step 2) Paste the embed code you copied from your Google Calendar below',
+								'jetpack'
+							) }
+						</p>
+						<form onSubmit={ this.setAttributes }>
```

Comment:
Is this form identical to the one in the inspector controls above? If so, we could abstract it to another function.

Reply:
good idea - have done that